src/vs/workbench/services/workingCopy/common/workingCopyFileService.ts

```diff
@@ -32,25 +32,23 @@ export interface WorkingCopyFileEvent extends IWaitUntil {
 	readonly operation: FileOperation;
 
 	/**
-	 * The resource the event is about.
+	 * The source resources that are defined for move operations.
+	 * The target resources the event is about.
 	 */
-	readonly target: URI;
-
-	/**
-	 * A property that is defined for move operations.
-	 */
-	readonly source?: URI;
+	readonly files: {
+		readonly source?: URI;
+		readonly target: URI;
+	}[]
 }
 
 export interface IWorkingCopyFileOperationParticipant {
 
 	/**
-	 * Participate in a file operation of a working copy. Allows to
-	 * change the working copy before it is being saved to disk.
+	 * Participate in a file operation of working copies. Allows to
+	 * change the working copies before they are being saved to disk.
 	 */
 	participate(
-		target: URI,
-		source: URI | undefined,
+		files: { source?: URI, target: URI }[],
```

Comment:
Can we extract this into a reusable type since we seem to be using that also above:

```ts
readonly files: {
   readonly source?: URI;
   readonly target: URI;
}[]
```

Reply:
I couldn't find a better name than `SourceTargetPair` type.